HotelBooking/HotelBooking.DataAccessLayer/Repositories/ReservationRepository.cs

```diff
@@ -8,8 +8,18 @@ internal class ReservationRepository : Repository<Reservation>, IReservationRepo
     public ReservationRepository(DbContext context) : base(context)
     { }
 
-    public IEnumerable<Reservation> GetReservationsFromDateRange(DateTime from, DateTime to)
+    public async Task<IEnumerable<Reservation>> GetReservationsFromDateRange(DateTime from, DateTime to)
     {
         return Context.Set<Reservation>().Where(r => DateTime.Compare(r.DateFrom, from) >=0 && DateTime.Compare(r.DateTo, to) <=0).OrderByDescending(r => r.ReservationId);
     }
+
+    public async Task<string> GetGuestFirstname(Reservation reservation)
```

Comment:
Te dwie metody nie powinny być w repozytorium Aniu. Zauważ, że Ty tutaj nic nie wyciągasz z bazy. Przekazujesz jako parametr obiekt rezerwacji który już mamy w pamięci i potem z niego wyciągasz konkretną właściwość. To nie jest rolą rezerwacji.